packages/caliper-core/lib/worker/tx-observers/prometheus-push-tx-observer.js

```diff
@@ -27,82 +31,142 @@ class PrometheusPushTxObserver extends TxObserverInterface {
-        this.prometheusClient.setGateway(options.push_url);
+        // do not use global registry to avoid conflicts with other potential prom-based observers
+        this.registry = new prometheus.Registry();
+
+        // automatically apply default internal and user supplied labels
+        this.defaultLabels = options.defaultLabels || {};
+        this.defaultLabels.workerIndex = this.workerIndex;
+        this.defaultLabels.roundIndex = this.currentRound;
+        this.defaultLabels.roundLabel = this.roundLabel;
+        this.registry.setDefaultLabels(this.defaultLabels);
+
+        // Exposed metrics
+        this.counterTxSubmitted = new prometheus.Counter({
+            name: 'caliper_tx_submitted',
+            help: 'The total number of submitted transactions.',
+            registers: [this.registry]
+        });
+
+        this.counterTxFinished = new prometheus.Counter({
+            name: 'caliper_tx_finished',
+            help: 'The total number of finished transactions.',
+            labelNames: ['final_status'],
+            registers: [this.registry]
+        });
+
+        // configure buckets
+        let buckets = prometheus.linearBuckets(0.1, 0.5, 10); // default
+        if (options.histogramBuckets) {
+            if (options.histogramBuckets.explicit) {
+                buckets = options.histogramBuckets.explicit;
+            } else if (options.histogramBuckets.linear) {
+                let linear = options.histogramBuckets.linear;
+                buckets = prometheus.linearBuckets(linear.start, linear.width, linear.count);
+            } else if (options.histogramBuckets.exponential) {
+                let exponential = options.histogramBuckets.exponential;
+                buckets = prometheus.exponentialBuckets(exponential.start, exponential.factor, exponential.count);
+            }
+        }
+
+        this.histogramLatency = new prometheus.Histogram({
+            name: 'caliper_tx_e2e_latency',
+            help: 'The histogram of end-to-end transaction latencies in seconds.',
+            labelNames: ['final_status'],
+            buckets,
+            registers: [this.registry]
+        });
+
+        // setting an interval enables the default metric collection
+        if (this.processMetricCollectInterval) {
+            this.processMetricHandle = prometheus.collectDefaultMetrics({
+                register: this.registry,
+                timestamps: false,
+                timeout: this.processMetricCollectInterval
+            });
+            const startGcStats = promGcStats(this.registry);
+            startGcStats();
+        }
 
-        this.internalStats = {
-            previouslyCompletedTotal: 0,
-            previouslySubmittedTotal: 0
-        };
+        const url = CaliperUtils.augmentUrlWithBasicAuth(options.pushUrl, Constants.AuthComponents.PushGateway);
+        this.prometheusClient = new prometheus.Pushgateway(url, null, this.registry);
```

Comment:
is this a prometheus push gateway client ? might be worth naming the variable explicitly to say that